Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift

```diff
@@ -88,6 +90,18 @@ internal class RUMViewScope: RUMScope, RUMContextProvider {
         self.viewStartTime = startTime
         self.dateCorrection = dependencies.dateCorrector.currentCorrection
         self.crashContextIntegration = RUMWithCrashContextIntegration()
+
+        if let cpuReader = RUMFeature.instance?.vitalCPUReader,
+           let memoryReader = RUMFeature.instance?.vitalMemoryReader,
+           let refreshRateReader = RUMFeature.instance?.vitalRefreshRateReader {
```

Comment:
We should rather inject it through `dependencies: RUMScopeDependencies` so it is easier to mock in tests and we do not depend on a global state that much.